src/parser/jobs/pld/modules/Gauge.tsx

```diff
@@ -0,0 +1,68 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import JOBS from 'data/JOBS'
+import {dependency} from 'parser/core/Module'
+import {CounterGauge, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
+import Suggestions, {SEVERITY, Suggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+// Gauge numbers
+const GAUGE_GAIN_AMOUNT = 5
+const GAUGE_SPEND_AMOUNT = 50
+
+export default class Gauge extends CoreGauge {
+	static title = t('pld.gauge.title')`Oath Gauge Usage`
+
+	@dependency private suggestions!: Suggestions
+
+	private oathGauge = this.add(new CounterGauge({
+		chart: {label: 'Oath Gauge', color: JOBS.PALADIN.colour},
+	}))
+
+	private oathConsumers = [
+		ACTIONS.SHELTRON.id,
+		ACTIONS.INTERVENTION.id,
+		ACTIONS.COVER.id,
+	]
+
+	protected init() {
+		super.init()
+
+		this.addEventHook('cast', {
+			by: 'player',
+			abilityId: ACTIONS.ATTACK.id,
+		}, this.onDamage)
+
+		this.addEventHook('cast', {
+			by: 'player',
+			abilityId: this.oathConsumers,
+		}, this.onConsumeOath)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	// HELPERS
+	private onDamage() {
+		this.oathGauge.modify(GAUGE_GAIN_AMOUNT)
+	}
+
+	private onConsumeOath() {
+		this.oathGauge.modify(-GAUGE_SPEND_AMOUNT)
+	}
```

Comment:
Do these really need to be separate functions? You can do this directly in the hook, with `() => this.oathGauge.modify(GAUGE_GAIN_AMOUNT)`.

Reply:
I don't like using nameless blocks tbh, that's why I used separate functions for each of them. I could change them if there's a convention that I must follow tho.